node/web/api/booksApiController.js

```diff
@@ -26,5 +26,72 @@ module.exports.factory = function (router, repo, exceptions) {
         });
     });
 
+     router.get('/api/book/addToCart/:uid', function (req, res) {
+        console.log(req.cookies.auth);
+         if (req.cookies.auth == undefined) {
+           res.send('noUser');
+           return;
+        }
+        else {
+                var cart;
+                console.log('server controller addtoCart get called');
+                if (req.cookies.cart === undefined) {
```

Comment:
While a cookie can be used to transmit data, such as the cart, it's not a good idea to do this. Cookies have all sorts of limitations that can be difficult to debug. Typically, we only use them for authentication purposes on page loads, as there are many features that help us avoid attacks, such as CSRF.

A better choice is to `POST`, `PUT` or `PATCH` the data to the API, where you would be able to access it on `req.body`. As a side note, you would typically not interact with `req.cookies.auth` in favor of `res.locals.user`, as the latter is guaranteed to be a `User` object, and the prior is not.